### ISSUE TITLE
Issue of weighted embeddings in Eq.(8)

### ISSUE THREAD

**Issue by HBinLi** [ninibymilk/PMF-MMEA#1]
Hi, thank you for your excellent work on this project! I’ve come across a potential issue in the PMF_Encoder.py file related to how weights are applied using the mask in the embeddings.
In particular, at line 42 of model/PMF_Encoder.py:
```python
embs = [F.normalize(emb * mask[key].unsqueeze(1).repeat(1, emb.shape[1])) for key, emb in embs_dict.items() if emb is not None]
```
It appears that while mask[key] correctly applies element-wise weights to the embeddings, the subsequent call to F.normalize scales the embeddings back to unit length. As a result, the effect of the weights is largely nullified unless mask[key] is exactly zero.
For values between 0 and 1 (e.g., 0.5 or 0.8), this normalization undoes the intended scaling, which reduces the impact of the weights and may lessen their effectiveness. This might not be the intended behavior in Eq.(8). That said, I might be misunderstanding something, and I’d really appreciate your thoughts on this. Looking forward to your response.
Thanks again for your great work!
